app/src/main/java/com/pileproject/drive/machine/CarControllerBase.java

```diff
@@ -0,0 +1,339 @@
+/**
+ * Copyright (C) 2011-2016 PILE Project, Inc. <dev@pileproject.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pileproject.drive.machine;
+
+import com.pileproject.drivecommand.machine.MachineBase;
+import com.pileproject.drivecommand.machine.device.input.ColorSensor;
+import com.pileproject.drivecommand.machine.device.input.GyroSensor;
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.Rangefinder;
+import com.pileproject.drivecommand.machine.device.input.RemoteControlReceiver;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Buzzer;
+import com.pileproject.drivecommand.machine.device.output.Led;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.machine.device.port.InputPort;
+import com.pileproject.drivecommand.machine.device.port.OutputPort;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.pileproject.drive.machine.CarControllerBase.MotorProperty.INIT_MOTOR_POWER;
+
+/**
+ * This is a base machine controller which controls a car-formed machine (e.g., Nxt/Ev3/Pile robot). A car-formed
+ * machine is a machine that at least has left and right {@link Motor}s. It depends on the machine what kinds of
+ * devices (e.g., {@link Buzzer}, {@link TouchSensor} it actually has, and such information (e.g.,
+ * methods/properties) should be treated in child controllers like {@link NxtCarController}.
+ *
+ * This base class must specify the interface to control a car-formed machine as methods which are the union of
+ * methods of child controllers, and provide the list of all input/output devices may be used (see: {@link InputDevice},
+ * {@link OutputDevice}).
+ * NOTE: Because of this, if you want add a new controller with a new feature which extends this base class , you
+ * should add a new one in this base class.
```

Comment:
"should add *the methods you defined in your new controller to* this base class"